prefer-object-rule false positives

I've just tried `eslint-plugin-eslint-plugin@4.0.0-0` in eslint repo.

```bash
/Users/weiran/repo/github/eslint/karma.conf.js
  13:18  error  Rules should be declared using the object style  eslint-plugin/prefer-object-rule

/Users/weiran/repo/github/eslint/lib/cli-engine/formatters/checkstyle.js
  31:18  error  Rules should be declared using the object style  eslint-plugin/prefer-object-rule
.....
```

We can suggest users to use `overrides` to avoid these cases?

--- a/README.md
+++ b/README.md
@@ -85,6 +85,23 @@ Presets are enabled by adding a line to the `extends` list in your config file. 
 }
 ```
 
+Or to apply linting only to the appropriate rule or test files:
+
+```json
+{
+    "overrides": [
+        {
+            "files": ["lib/rules/*.{js,ts}"],
+            "extends": ["plugin:eslint-plugin/rules-recommended"]
+        },
+        {
+            "files": ["tests/lib/rules/*.{js,ts}"],
+            "extends": ["plugin:eslint-plugin/tests-recommended"]
+        },
+    ]
+}
+```
+
 * `recommended` enables all recommended rules from this plugin.
 * `rules-recommended` enables all recommended rules that are aimed at linting ESLint rule files.
 * `tests-recommended` enables all recommended rules that are aimed at linting ESLint test files.

